src/modules/musig/session_impl.h

```diff
@@ -0,0 +1,755 @@
+    secp256k1_gej aggnonce_ptj[2];
+    secp256k1_ge aggnonce_pt[2];
+    int i;
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(aggnonce != NULL);
+    ARG_CHECK(pubnonces != NULL);
+    ARG_CHECK(n_pubnonces > 0);
+
+    if (!secp256k1_musig_sum_nonces(ctx, aggnonce_ptj, pubnonces, n_pubnonces)) {
+        return 0;
+    }
+    for (i = 0; i < 2; i++) {
+        if (secp256k1_gej_is_infinity(&aggnonce_ptj[i])) {
+            /* There must be at least one dishonest signer. If we would return 0
+               here, we will never be able to determine who it is. Therefore, we
+               should continue such that the culprit is revealed when collecting
+               and verifying partial signatures.
+
+               However, dealing with the point at infinity (loading,
+               de-/serializing) would require a lot of extra code complexity.
+               Instead, we set the aggregate nonce to some arbitrary point (the
+               generator). This is secure, because it only restricts the
+               abilities of the attacker: an attacker that forces the sum of
+               nonces to be infinity by sending some maliciously generated nonce
+               pairs can be turned into an attacker that forces the sum to be
+               the generator (by simply adding the generator to one of the
+               malicious nonces), and this does not change the winning condition
+               of the EUF-CMA game. */
+            aggnonce_pt[i] = secp256k1_ge_const_g;
+        } else {
+            secp256k1_ge_set_gej(&aggnonce_pt[i], &aggnonce_ptj[i]);
+        }
+    }
+    secp256k1_musig_aggnonce_save(aggnonce, aggnonce_pt);
+    return 1;
+}
+
+/* tagged_hash(aggnonce[0], aggnonce[1], agg_pk, msg) */
+static int secp256k1_musig_compute_noncehash(unsigned char *noncehash, secp256k1_ge *aggnonce, const unsigned char *agg_pk32, const unsigned char *msg) {
+    unsigned char buf[33];
+    secp256k1_sha256 sha;
+    int i;
+
+    secp256k1_sha256_initialize_tagged(&sha, (unsigned char*)"MuSig/noncecoef", sizeof("MuSig/noncecoef") - 1);
+    for (i = 0; i < 2; i++) {
+        size_t size;
+        if (!secp256k1_eckey_pubkey_serialize(&aggnonce[i], buf, &size, 1)) {
+            return 0;
+        }
+        VERIFY_CHECK(size == sizeof(buf));
+        secp256k1_sha256_write(&sha, buf, sizeof(buf));
+    }
+    secp256k1_sha256_write(&sha, agg_pk32, 32);
+    secp256k1_sha256_write(&sha, msg, 32);
+    secp256k1_sha256_finalize(&sha, noncehash);
+    return 1;
+}
+
+static int secp256k1_musig_nonce_process_internal(int *fin_nonce_parity, unsigned char *fin_nonce, secp256k1_scalar *b, secp256k1_gej *aggnoncej, const unsigned char *agg_pk32, const unsigned char *msg) {
+    unsigned char noncehash[32];
+    secp256k1_ge fin_nonce_pt;
+    secp256k1_gej fin_nonce_ptj;
+    secp256k1_ge aggnonce[2];
+
+    secp256k1_ge_set_gej(&aggnonce[0], &aggnoncej[0]);
+    secp256k1_ge_set_gej(&aggnonce[1], &aggnoncej[1]);
```

Comment:
Note that the function `secp256k1_musig_nonce_process` already has computed the secp256k1_ge values for aggnonce, it might be worth passing them directly to the `_internal` function instead of computing them again.

Reply:
Indeed. The [0] value may have changed though when there's an aggregator. But we could still pass the ge values to `_internal`. Do you want to open a PR?